csi/volumeutils.py

```diff
@@ -850,38 +850,39 @@ def volume_list(voltype=None):
 def mount_volume(pvpath, mountpoint, pvtype, fstype=None):
     """Mount a Volume"""
 
-    # Create subvol dir if PV is manually created
-    if not os.path.exists(pvpath):
-        makedirs(pvpath)
-
-    # TODO: Will losetup survive container reboot?
-    if pvtype == PV_TYPE_RAWBLOCK:
-        # losetup of truncated file
-        cmd = ["losetup", "-f", "--show", pvpath]
-        try:
-            loop, _, _ = execute(*cmd)
-        except CommandException as err:
-            # Better not to create loop devices manually
-            errmsg = "Please check availability of 'losetup' and 'loop' device"
-            logging.error(logf(errmsg, cmd=cmd, error=format(err)))
-            return False
-
-        # Bind mount loop device to target_path, stage_path may not be needed
-        makedirs(os.path.dirname(mountpoint))
-        Path(mountpoint).touch(mode=0o777)
-        execute(MOUNT_CMD, "--bind", loop, mountpoint)
-        return True
-
-    # Need this after kube 1.20.0
-    makedirs(mountpoint)
-
-    if pvtype == PV_TYPE_VIRTBLOCK:
-        fstype = "xfs" if fstype is None else fstype
-        execute(MOUNT_CMD, "-t", fstype, pvpath, mountpoint)
-    else:
-        execute(MOUNT_CMD, "--bind", pvpath, mountpoint)
+    if not os.path.ismount(mountpoint):
```

Comment:
Instead of adding everything in if block, return early if mount exists.

```python
if os.path.ismount(mountpoint):
    return True
```

Reply:
Good point. Done!